pkg/types/nutanix/validation/platform.go

```diff
@@ -0,0 +1,85 @@
+package validation
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/installer/pkg/types/nutanix"
+	"github.com/openshift/installer/pkg/validate"
+)
+
+// ValidatePlatform checks that the specified platform is valid.
+// TODO(nutanix): Revisit for further expanding the validation logic
+func ValidatePlatform(p *nutanix.Platform, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(p.PrismCentral) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("prismCentral"), "must specify the Prism Central"))
+	}
+	if len(p.PrismElement) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("prismElement"), "must specify the Prism Element"))
+	}
+	if len(p.Username) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("username"), "must specify the username"))
+	}
+	if len(p.Password) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("password"), "must specify the password"))
+	}
+	if len(p.Port) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("port"), "must specify the port"))
+	}
+	if len(p.Subnet) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("subnet"), "must specify the subnet"))
+	}
+	if len(p.PrismCentral) != 0 {
+		if err := validate.Host(p.PrismCentral); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("prismCentral"), p.PrismCentral, "must be the domain name or IP address of the Prism Central"))
+		}
+	}
+
+	// If all VIPs are empty, skip IP validation.  All VIPs are required to be defined together.
+	if strings.Join([]string{p.APIVIP, p.IngressVIP}, "") != "" {
+		allErrs = append(allErrs, validateVIPs(p, fldPath)...)
+	}
+
+	return allErrs
+}
+
+// ValidateForProvisioning checks that the specified platform is valid.
+func ValidateForProvisioning(p *nutanix.Platform, fldPath *field.Path) field.ErrorList {
```

Comment:
This function should be validating that resources that need to exist in the Nutanix provider exist and are configured as expected. This function is not a place to do static validation checking.

Reply:
Fair point. We will create future PRs that will switch the static validation to resource validation i.e. see if a prism element with the UUID actually exists on a prism central cluster.

Reply:
Remove the static validation, as there is no value in having it in here now. The future PRs can add resource validation, but they do not need to replace static validation.

Reply:
@staebler Looking at the validation package for different cloud providers, most of them are also doing just static validation.

Reply:
Removed `ValidateForProvisioning`